src/rtnl/link/nlas/link_xdp.rs

```diff
@@ -0,0 +1,322 @@
+// SPDX-License-Identifier: MIT
+
+use std::{convert::TryFrom, mem::size_of, os::fd::RawFd};
+
+use anyhow::Context;
+use byteorder::{NativeEndian, WriteBytesExt};
+use utils::parsers::parse_i32;
+
+use crate::{
+    constants::*,
+    nlas::{Nla, NlaBuffer, NlasIterator},
+    parsers::parse_u32,
+    traits::Parseable,
+    DecodeError,
+};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Xdp {
+    Unspec(Vec<u8>),
+    Fd(RawFd),
+    Attached(XdpAttached),
+    Flags(u32),
+    ProgId(u32),
+    DrvProgId(u32),
+    SkbProgId(u32),
+    HwProgId(u32),
+    ExpectedFd(u32),
+}
+
+impl Nla for Xdp {
+    #[rustfmt::skip]
+    fn value_len(&self) -> usize {
+        use self::Xdp::*;
+        match self {
+            Unspec(ref bytes)
+                => bytes.len(),
+            Fd(_) => size_of::<RawFd>(),
+            Attached(_) => size_of::<XdpAttached>(),
+            Flags(_) => size_of::<u32>(),
+            ProgId(_) => size_of::<u32>(),
+            DrvProgId(_) => size_of::<u32>(),
+            SkbProgId(_) => size_of::<u32>(),
+            HwProgId(_) => size_of::<u32>(),
+            ExpectedFd(_) => size_of::<u32>(),
+        }
+    }
+
+    #[rustfmt::skip]
+    fn emit_value(&self, mut buffer: &mut [u8]) {
+        use self::Xdp::*;
+        match self {
+            Unspec(ref bytes)
+                => buffer.copy_from_slice(bytes),
+            Fd(ref value) => buffer.write_i32::<NativeEndian>(*value).unwrap(),
+            Attached(ref value) => buffer.write_u8(*value as u8).unwrap(),
+            Flags(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
+            ProgId(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
+            DrvProgId(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
+            SkbProgId(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
+            HwProgId(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
+            ExpectedFd(ref value) => buffer.write_u32::<NativeEndian>(*value).unwrap(),
```

Comment:
`unwrap()` is not allowed in crates as it will panic the program.

Meanwhile, use `NativeEndian::write_u32()`.